sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -1250,7 +1253,10 @@ pub fn oci_defaults(
             runtime.get_capabilities(capabilities)
         }
         OciSpecSection::ResourceLimits => {
-            let rlimits = oci_spec_resource_limits(oci_defaults_values)?;
+            let mut rlimits = oci_spec_resource_limits(oci_defaults_values)?;
+            if pciclient::is_efa_attached().context(error::CheckEfaFailureSnafu)? {
+                adjust_resource_limits_for_efa(&mut rlimits);
+            }
```

Comment:
Ah, I think these unit tests are failing. Exposing the generic `lspci` backend to consumers so they can write tests might be helpful (or implementing a similar interface-based backend here)

Reply:
Right, I moved the detect EFA block out of the `oci_spec_resource_limits`, now the unit tests are passing. The `oci_defaults` is not included in the unit tests so putting this logic here is not breaking any tests. I also made sure the `adjust_resource_limits_for_efa` logic is tested (added a test case for it).

Reply:
Right now the CI is not happy with the doctest, I will modify the docstring so that only the rust code example can be ran.